tracer/src/Datadog.Trace/Telemetry/Metrics/PublicApiUsage.cs

```diff
@@ -23,7 +23,13 @@ internal enum PublicApiUsage
     [Description("name:eventtrackingsdk_trackuserloginsuccessevent_metadata")]EventTrackingSdk_TrackUserLoginSuccessEvent_Metadata,
 
     [Description("name:spancontextextractor_extract")] SpanContextExtractor_Extract,
+
+    [Description("name:spancontextextractor_extractincludingdsm")]
+    SpanContextExtractor_ExtractIncludingDsm,
```

Comment:
These don't appear to be used - it appears you're missing the auto-instrumentation types? 🤔

Reply:
hmm, I think I don't know what that auto-instrumentation is supposed to do.
I my mind, I was just adding a method that users can call manually and that's it, nothing `auto` there

Reply:
The custom will call the `ExtractWithDsm()` method which is in Datadog.Trace.Manual. However, that method currently doesn't _do_ anything (it always returns `null`). The automatic instrumentation needs to re-write that method so that it will call the actual `SpanContextExtractor` method in Datadog.Trace.

Basically, you need an equivalent of this: 

https://github.com/DataDog/dd-trace-dotnet/blob/0963e70c7ba853df64bd143ea836d518492cf11c/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/ManualInstrumentation/Propagators/SpanContextExtractorExtractIntegration.cs#L32

If that doesn't make sense happy to chat about it tomorrow 🙂

Reply:
OK, I'm ooo tomorrow, but I'll look into that when I come back on Tuesday